pkg/app/server/apikeyverifier/verifier.go

```diff
@@ -54,37 +55,41 @@ func (v *Verifier) Verify(ctx context.Context, key string) (*model.APIKey, error
 	item, err := v.apiKeyCache.Get(keyID)
 	if err == nil {
 		apiKey = item.(*model.APIKey)
-		if err := checkAPIKey(apiKey, keyID, key); err != nil {
+		if err := checkAPIKey(ctx, v, apiKey, keyID, key); err != nil {
 			return nil, err
 		}
 		return apiKey, nil
 	}
-
 	// If the cache data was not found,
 	// we have to retrieve from datastore and save it to the cache.
 	apiKey, err = v.apiKeyStore.Get(ctx, keyID)
 	if err != nil {
 		return nil, fmt.Errorf("unable to find API key %s from datastore, %w", keyID, err)
 	}
 
+	// update the time API key was last used
+
 	if err := v.apiKeyCache.Put(keyID, apiKey); err != nil {
 		v.logger.Warn("unable to store API key in memory cache", zap.Error(err))
 	}
-	if err := checkAPIKey(apiKey, keyID, key); err != nil {
+	if err := checkAPIKey(ctx, v, apiKey, keyID, key); err != nil {
 		return nil, err
 	}
 
 	return apiKey, nil
 }
 
-func checkAPIKey(apiKey *model.APIKey, id, key string) error {
+func checkAPIKey(ctx context.Context, v *Verifier, apiKey *model.APIKey, id, key string) error {
```

Comment:
If we need to pass the Verifier as a parameter here, then better to make this checkAPIKey as a private method of the Verifier struct.